docs/docs/concepts/server-mcp.md

```diff
@@ -0,0 +1,187 @@
+
+[LangGraph Server](./langgraph_server.md) implements MCP using the [Streamable HTTP transport](https://spec.modelcontextprotocol.io/specification/2025-03-26/basic/transports/#streamable-http). This allows LangGraph **agents** to be exposed as **MCP tools**, making them usable with any MCP-compliant client supporting Streamable HTTP.
+
+The MCP endpoint is available at:
+
+```
+/mcp
+```
+
+on [LangGraph Server](./langgraph_server.md).
+
+## Requirements
+
+To use MCP, ensure you have the following dependencies installed:
+
+- `langgraph-api >= 0.2.3`
+- `langgraph-sdk >= 0.1.61`
+
+Install them with:
+
+```bash
+pip install "langgraph-api>=0.2.3" "langgraph-sdk>=0.1.61"
+```
+
+## Exposing an agent as MCP tool
+
+
+When deployed, your agent will appear as a tool in the MCP endpoint
+with this configuration:
+
+- **Tool name**: The agent's name.
+- **Tool description**: The agent's description.
+- **Tool input schema**: The agent's input schema.
+
+### Setting name and description 
+
+You can set the name and description of your agent in `langgraph.json`:
+
+```json
+{
+    "graphs": {
+        "my_agent": {
+            "path": "./my_agent/agent.py:graph",
+            "description": "A description of what the agent does"
+        }
+    },
+    "env": ".env"
+}
+```
+
+After deployment, you can update the name and description using the LangGraph SDK.
+
+### Schema
+
+Define clear, minimal input and output schemas to avoid exposing unnecessary internal complexity to the LLM.
+
+The default [MessagesState](./low_level.md#messagesstate) uses `AnyMessage`, which supports many message types but is too general for direct LLM exposure.
+
+Instead, define **custom agents or workflows** that use explicitly typed input and output structures.
+
+For example, a workflow answering documentation questions might look like this:
+
+```python
+from langgraph.graph import StateGraph, START, END
+from typing_extensions import TypedDict
+
+# Define input schema
+class InputState(TypedDict):
+    question: str
+
+# Define output schema
+class OutputState(TypedDict):
+    answer: str
+
+# Combine input and output
+class OverallState(InputState, OutputState):
+    pass
+
+# Define the processing node
+def answer_node(state: InputState):
+    # Replace with actual logic and do something useful
+    return {"answer": "bye", "question": state["question"]}
+
+# Build the graph with explicit schemas
+builder = StateGraph(OverallState, input=InputState, output=OutputState)
+builder.add_node(answer_node)
+builder.add_edge(START, "answer_node")
+builder.add_edge("answer_node", END)
+graph = builder.compile()
+
+# Run the graph
+print(graph.invoke({"question": "hi"}))
+```
+
+For more details, see the [low-level concepts guide](https://langchain-ai.github.io/langgraph/concepts/low_level/#state).
+
+
+## Usage overview
+
+To enable MCP:
+
+- Upgrade to a recent LangGraph server version.
```

Comment:
Upgrade to use `langgraph-api>=0.2.3`. If you are deploying on LangGraph Platform, this will likely be done for you if you create a new revision.